borg/repository.py

```diff
@@ -308,6 +317,8 @@ def complete_xfer():
         self.compact = set()
 
     def replay_segments(self, index_transaction_id, segments_transaction_id):
+        # fake an old client, so that in case we do not have an exclusive lock yet, prepare_txn will upgrade the lock:
+        self.exclusive = None
```

Comment:
Hm, we do a rollback() in the `finally` clause, so a new transaction could be initiated by the client after replay_segments completes. Not sure if we do that anywhere, I don't think so.